azurerm/internal/services/network/firewall_policy_resource.go

```diff
@@ -0,0 +1,356 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmFirewallPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmFirewallPolicyCreateUpdate,
+		Read:   resourceArmFirewallPolicyRead,
+		Update: resourceArmFirewallPolicyCreateUpdate,
+		Delete: resourceArmFirewallPolicyDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.FirewallPolicyID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FirewallPolicyName(),
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": location.Schema(),
+
+			"base_policy_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.FirewallPolicyID,
+			},
+
+			"dns_setting": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"servers": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validation.IsIPv4Address,
+							},
+						},
+						"proxy_enabled": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+						"network_rules_fqdn_enabled": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+					},
+				},
+			},
+
+			"threat_intelligence_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  string(network.AzureFirewallThreatIntelModeAlert),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(network.AzureFirewallThreatIntelModeAlert),
+					string(network.AzureFirewallThreatIntelModeDeny),
+					string(network.AzureFirewallThreatIntelModeOff),
+				}, false),
+			},
+
+			"threat_intelligence_whitelist": {
```

Comment:
In the spirit of inclusive language we are trying to avoid terms like white/black list in the provider. As such could we change this to
```suggestion
			"threat_intelligence_filter": {
```
or
```suggestion
			"threat_intelligence_options": {
```

or something else that makes sense

Reply:
Oh, didn't think of that point! Thank you for reminding me!

Reply:
Based on [this](https://tools.ietf.org/id/draft-knodel-terminology-00.html#suggested-alternatives-1), I'm gonna rename it to `threat_intelligence_ignorelist`.

Reply:
whilst that is the direct replacement for that term - i'm not sure it makes as much sense as filter? given i think the SDK/API terminology was a poor choice to start with. These are i presume values to filter on? so `threat_intelligence_filters` sounds more correct & natural then `threat_intelligence_allowlist` as it is a accepted and widely used term, ie "filter the log on these values"

Reply:
In this case, the ip/fqdn specified in the block means those ip/fqdn are to be ignored for threat intelligence detection. While "filter" sounds like the opposite meaning (i.e. instead of ignoring them, sounds like to only process them). Using "allowlist" here sounds more correct as it means to allow/tolerate the specified ip/fqdn.
What do you think?

Reply:
I would have gone with "threat_intelligence_whitelist" to match the associated Microsoft terminology. But agree "threat_intelligence_allowlist" is still good.

Reply:
that makes sense to me!